src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/TaskControllerTest.java

```diff
@@ -128,6 +142,68 @@ void should_return_a_500_when_id_invalid() throws Exception {
             verify(camundaServiceApi, times(1))
                 .getTask(any(), any());
         }
+
+
+        @Test
+        void should_return_a_400_when_restricted_role_is_given() throws Exception {
+            final var taskId = UUID.randomUUID().toString();
+
+            final var userToken = "user_token";
+
+            mockServices.mockUserInfo(idamWebApi);
+
+            final List<String> roleNames = asList("tribunal-caseworker");
+
+            Map<String, String> roleAttributes = new HashMap<>();
+            roleAttributes.put(RoleAttributeDefinition.JURISDICTION.value(), "IA");
+
+            // Role attribute is IA
+            List<Assignment> allTestRoles = new ArrayList<>();
+            roleNames.forEach(roleName -> asList(RoleType.ORGANISATION, RoleType.CASE)
+                .forEach(roleType -> {
+                    Assignment roleAssignment = mockServices.createBaseAssignment(
+                        UUID.randomUUID().toString(), "tribunal-caseworker",
+                        roleType,
+                        Classification.PUBLIC,
+                        roleAttributes
+                    );
+                    allTestRoles.add(roleAssignment);
+                }));
+
+            GetRoleAssignmentResponse accessControlResponse = new GetRoleAssignmentResponse(
+                allTestRoles
+            );
+            when(roleAssignmentServiceApi.getRolesForUser(
+                any(), any(), any()
+            )).thenReturn(accessControlResponse);
+
+            when(idamWebApi.token(any())).thenReturn(new Token(userToken, "scope"));
+
+            // Task created with Jurisdiction SCSS
+            mockCamundaVariables();
+
+            mockMvc.perform(
+                get("/task/" + taskId)
+                    .header(
+                        "Authorization",
+                        authorizationHeadersProvider.getTribunalCaseworkerAAuthorization()
+                    )
+                    .contentType(MediaType.APPLICATION_JSON_VALUE)
+            ).andExpect(status().is4xxClientError());
```

Comment:
we should probably change this to be more specific I believe in this particular scenario we want this to fail as the user didn't have enough permissions to the task as he was in a different jurisdiction so in this case we expect a 403.

Reply:
Agreed.